sdks/python/apache_beam/runners/common.py

```diff
@@ -1941,3 +1945,64 @@ def validate_transform(transform_id):
 
   for t in pipeline_proto.root_transform_ids:
     validate_transform(t)
+
+
+def merge_common_environments(pipeline_proto):
```

Comment:
Does this make the merge logic at the following location obsolete ?

https://github.com/apache/beam/blob/fb7ba65e2236f3dd871b6e492afc07249a4a5c49/sdks/python/apache_beam/pipeline.py#L964

Reply:
Ah, yes, it looks like it does. (That code didn't seem to be working, as I was definitely seeing environments that needed deduplication, but perhaps I should merge the two.)

Reply:
Consolidated. It's still good to have it in both places.